### DISCUSSION TITLE
v1.56.2 docker image based on alpine rather than debian

### DISCUSSION THREAD

**Discussion by peterstace** [golangci/golangci-lint#4391] [Q&A] [answered]
Previous docker images (e.g. [`v1.56.1`](https://hub.docker.com/layers/golangci/golangci-lint/v1.56.1/images/sha256-f7bfcbe7b8f9ef6c7a7b06f372c66f8cdfbbba620837ab1e3bb7bf27079a30a2?context=explore)) were based on debian. However, the new [`v1.56.2`](https://hub.docker.com/layers/golangci/golangci-lint/v1.56.2/images/sha256-0a857e707833f61a5782a95eb17db0d0af4482e133a855077a331fa46e59b044?context=explore) image is based on alpine.

This is breaking a few things for me (e.g. installing extra packages with `apt-get`).

So I'm wondering:

- Is this an intentional change?
- Will future docker images be based on debian or alpine?
- No drama at all if switching to alpine was intentional (it's very easy for me to switch, I just don't want to switch if this was an accidental change that will be reverted in the next tag).

**Comment by peterstace** [comment]
This _might_ be related to https://github.com/golangci/golangci-lint/pull/4370? I'm not too sure; that PR modified the Dockerfile, but it still looks debian-based (since the base image of [buildx.Dockerfile](https://github.com/golangci/golangci-lint/blob/master/build/buildx.Dockerfile) is still `golang:1.22`.

**Reply by ldez** [reply]
:thinking: the base images inside Dockerfile are still the same, I will investigate.

**Reply by ldez** [reply]
Not all tags are affected :thinking: 

```console
$ docker run --rm -it golangci/golangci-lint:v1.56 sh  
# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
# golangci-lint version
golangci-lint has version 1.56.2 built with go1.22.0 from 58a724a0 on 2024-02-15T18:01:51Z
# 
```

```console
$ docker run --rm -it golangci/golangci-lint:latest sh
# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
# golangci-lint version
golangci-lint has version 1.56.2 built with go1.22.0 from 58a724a0 on 2024-02-15T18:01:51Z
# 
```

```console
$ docker run --rm -it golangci/golangci-lint:v1.56.2 sh
/app # cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.19.1
PRETTY_NAME="Alpine Linux v3.19"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"
/ # golangci-lint version
golangci-lint has version 1.56.2 built with go1.22.0 from 58a724a0 on 2024-02-15T18:01:51Z
/ # 
```

**Comment by ldez** [comment, marked as answer]
I found the issue. https://github.com/golangci/golangci-lint/pull/4392

I re-published the Docker images, the problem with the v1.56.2 tag is fixed.

**Reply by ldez** [reply]
I re-published the images.

**Reply by ldez** [reply]
Thank you for the report @peterstace :heart:

**Reply by peterstace** [reply]
Thanks! I've confirmed that the republished images are working for me.
